### PR TITLE
Update react-router-dom NavLinkProps according to v5.3.0 changes

### PR DESCRIPTION
**Pull request by MikeSha**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/remix-run/react-router/releases/tag/v5.3.0>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by eps1lon — CHANGES_REQUESTED** [review]
Thanks for the work.

Please add a test to make sure we don't regress in the future.

File: types/react-router-dom/index.d.ts [hunk 1]
```diff
@@ -71,13 +71,17 @@ export interface Link<S = H.LocationState>
...
 
-export interface NavLinkProps<S = H.LocationState> extends LinkProps<S> {
+export interface NavLinkProps<S = H.LocationState> extends Omit<LinkProps<S>, "className" | "style"> {
     activeClassName?: string | undefined;
     activeStyle?: React.CSSProperties | undefined;
     exact?: boolean | undefined;
     strict?: boolean | undefined;
     isActive?<Params extends { [K in keyof Params]?: string }>(match: match<Params> | null, location: H.Location<S>): boolean;
     location?: H.Location<S> | undefined;
+    className?: string | ((isActive: boolean) => string);
```
Comment (MikeSha) [inline types/react-router-dom/index.d.ts:81]: According to the docs it should be `(props: { isActive: boolean })`, but in the reality it is `(isActive: boolean)`
I'm not sure if this PR should be merged before `react-router` devs fix this issue

File: types/react-router-hash-link/index.d.ts [hunk 2]
```diff
@@ -16,7 +16,10 @@ export interface HashLinkProps extends LinkProps {
   timeout?: number | undefined;
 }
 
-export interface NavHashLinkProps extends NavLinkProps, HashLinkProps { }
+export interface NavHashLinkProps extends Omit<NavLinkProps, 'className' | 'style'>, HashLinkProps {
```
Comment (MikeSha) [inline types/react-router-hash-link/index.d.ts:19]: This package doesn't accept function in `className` and `style`

File: types/react-router-dom/index.d.ts (hunk 1 above)
Comment (eps1lon) [reply types/react-router-dom/index.d.ts:81]: > According to the docs

Could you link the docs? I only found mention of this feature in the release notes.

We should sync with the react-router folks on what the intended interface is here. It's a bit odd that we document different types than are actually documented.

Comment (MikeSha) [reply types/react-router-dom/index.d.ts:81]: Seems to be it's the only mention of this change. Anyways, I've created an issue https://github.com/remix-run/react-router/issues/8028. Lets wait until they decide how it should be fixed and then I'll update this PR if necessary

Comment (MikeSha) [reply types/react-router-dom/index.d.ts:81]: @eps1lon https://github.com/remix-run/react-router/issues/8028#issuecomment-915212576 Seems to be this PR could be merged :)

**Review by eps1lon — APPROVED** [review]

**Comment by MikeSha** [comment]
Ready to merge

**Pull request merged**